src/tweenable.rs

```diff
@@ -894,264 +917,399 @@ mod tests {
+            directions: repeat(direction),
+            states: repeat(TweenState::Active)
+                .take(5)
+                .chain(repeat(TweenState::Completed)),
+            transforms: if direction == TweeningDirection::Forward {
+                successors(Some(0.), |progress| Some(f32::min(progress + 0.2, 1.)))
+                    .map(|progress| Transform::from_translation(Vec3::splat(progress)))
+                    .take(7)
+                    .collect::<Vec<_>>()
+            } else {
+                successors(Some(1.), |progress| Some(f32::max(progress - 0.2, 0.)))
+                    .map(|progress| Transform::from_translation(Vec3::splat(progress)))
+                    .take(7)
+                    .collect::<Vec<_>>()
+            },
+        };
+
+        validate_tween(tween, expected_values);
+    }
+
+    #[test]
+    fn tween_tick_loop_finite() {
+        let tween = Tween::new(
+            EaseMethod::Linear,
+            Duration::from_secs(1),
+            TransformPositionLens {
+                start: Vec3::ZERO,
+                end: Vec3::ONE,
+            },
+        )
+        .with_direction(TweeningDirection::Forward)
+        .with_repeat_count(RepeatCount::Finite(3))
+        .with_repeat_strategy(RepeatStrategy::Repeat);
+
+        let expected_values = ExpectedValues {
+            deltas: once(Duration::ZERO).chain(repeat(Duration::from_secs_f32(1. / 3.)).take(10)),
+            progress: successors(Some(0.), |progress| Some(f32::min(3., progress + 1. / 3.))),
+            // TODO this is totally wrong due to float precision, should be fixed in upcoming
+            //  PR
+            times_completed: [0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3],
+            event_counts: [0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3],
+            directions: repeat(TweeningDirection::Forward),
+            states: repeat(TweenState::Active)
+                .take(9)
+                .chain(repeat(TweenState::Completed)),
+            transforms: successors(Some(0.), |progress| Some(f32::min(3., progress + 1. / 3.)))
+                .map(|progress| Transform::from_translation(Vec3::splat(progress))),
+        };
+
+        validate_tween(tween, expected_values);
+    }
+
+    #[test]
+    fn tween_tick_loop_infinite_large_jump() {
+        let duration = Duration::from_secs_f64(4. / 3.);
+        let completions = Duration::MAX.as_secs_f64() / duration.as_secs_f64();
+        let tween = Tween::new(
+            EaseMethod::Linear,
+            duration,
+            TransformPositionLens {
+                start: Vec3::ZERO,
+                end: Vec3::ONE,
+            },
+        )
+        .with_direction(TweeningDirection::Forward)
+        .with_repeat_count(RepeatCount::Infinite)
+        .with_repeat_strategy(RepeatStrategy::Repeat);
+
+        let expected_values = ExpectedValues {
+            deltas: [duration * 10, Duration::MAX, Duration::MAX],
+            progress: once(10.).chain(repeat(completions as f32)),
```

Comment:
Note we have the same issue here, since `duration`  is 4/3 so irrational, we have no strong guarantee that 10 iterations land _exactly_ on the completed state (short of diving into IEEE and proving it for that particular case, but I want to avoid that).

Reply:
Fixed.